File: src/config/platform2Categories.js
/**
 * Platform 2 Categories Configuration
 * Defines the categories, tiers, and grid positions for the 5x5 layout.
 * Updated to use FontAwesome icons for a more professional look.
 */

export const PLATFORM2_CATEGORIES = [
    // Row 1: High Value (Tier 1)
    {
        id: "real-estate",
        name: "Real Estate & Property Agents",
        multiplier: 2.5,
        tier: "tier1",
        iconClass: "fa-solid fa-house-chimney",
        gridPos: 1
    },
    {
        id: "legal-services",
        name: "Legal Services, Lawyers & Mediation",
        multiplier: 2.5,
        tier: "tier1",
        iconClass: "fa-solid fa-scale-balanced",
        gridPos: 2
    },
    {
        id: "financial-services",
        name: "Financial and Insurance Services",
        multiplier: 2.3,
        tier: "tier1",
        iconClass: "fa-solid fa-building-columns",
        gridPos: 3
    },
    {
        id: "health-medical",
        name: "Health, Wellness & Medical",
        multiplier: 2.3,
        tier: "tier1",
        iconClass: "fa-solid fa-house-medical",
        gridPos: 4
    },
    {
        id: "automotive-services",
        name: "Automotive Services",
        multiplier: 2.2,
        tier: "tier1",
        iconClass: "fa-solid fa-screwdriver-wrench",
        gridPos: 5
    },
    // Row 2: High Value & Top Medium Value
    {
        id: "it-tech",
        name: "IT & Tech Support Services",
        multiplier: 2.0,
        tier: "tier1",
        iconClass: "fa-solid fa-laptop-code",
        gridPos: 6
    },
    {
        id: "professional-training",
        name: "Professional Training & Certification",
        multiplier: 2.0,
        tier: "tier1",
        iconClass: "fa-solid fa-user-graduate",
        gridPos: 7
    },
    {
        id: "electronics",
        name: "Department Stores and Electronics",
        multiplier: 2.0,
        tier: "tier1",
        iconClass: "fa-solid fa-tv",
        gridPos: 8
    },
    {
        id: "mobile-internet",
        name: "Mobile Phone and Internet Services",
        multiplier: 2.0,
        tier: "tier1",
        iconClass: "fa-solid fa-mobile-screen-button",
        gridPos: 9
    },
    {
        id: "education-tutoring",
        name: "Education & Tutoring",
        multiplier: 1.8,
        tier: "tier2",
        iconClass: "fa-solid fa-book-open-reader",
        gridPos: 10
    },
    // Row 3: Medium Value
    {
        id: "event-wedding",
        name: "Event & Wedding Services",
        multiplier: 1.8,
        tier: "tier2",
        iconClass: "fa-solid fa-champagne-glasses",
        gridPos: 11
    },
    {
        id: "beauty-surgery",
        name: "Beauty and Cosmetic Surgery",
        multiplier: 1.7,
        tier: "tier2",
        iconClass: "fa-solid fa-wand-sparkles",
        gridPos: 12
    },
    {
        id: "fitness-training",
        name: "Fitness & Personal Training",
        multiplier: 1.6,
        tier: "tier2",
        iconClass: "fa-solid fa-dumbbell",
        gridPos: 13
    },
    {
        id: "home-garden",
        name: "Home & Garden",
        multiplier: 1.6,
        tier: "tier2",
        iconClass: "fa-solid fa-leaf",
        gridPos: 14
    },
    {
        id: "lifestyle-apparel",
        name: "Lifestyle, Boutique, Apparel & Accessories",
        multiplier: 1.5,
        tier: "tier2",
        iconClass: "fa-solid fa-shirt",
        gridPos: 15
    },
    // Row 4: Medium & Entry Level
    {
        id: "travel-tourism",
        name: "Travel & Tourism",
        multiplier: 1.5,
        tier: "tier2",
        iconClass: "fa-solid fa-plane-departure",
        gridPos: 16
    },
    {
        id: "storage-logistics",
        name: "Storage, Logistics and Removalists",
        multiplier: 1.5,
        tier: "tier2",
        iconClass: "fa-solid fa-truck-ramp-box",
        gridPos: 17
    },
    {
        id: "restaurants-fnb",
        name: "Restaurants, Food & Beverage",
        multiplier: 1.3,
        tier: "tier3",
        iconClass: "fa-solid fa-utensils",
        gridPos: 18
    },
    {
        id: "trades-services",
        name: "Trades & Home Services",
        multiplier: 1.3,
        tier: "tier3",
        iconClass: "fa-solid fa-hammer",
        gridPos: 19
    },
    {
        id: "pets-animals",
        name: "Pets & Animals",
        multiplier: 1.2,
        tier: "tier3",
        iconClass: "fa-solid fa-paw",
        gridPos: 20
    },
    // Row 5: Entry Level
    {
        id: "childcare-agedcare",
        name: "Childcare & Aged Care Providers",
        multiplier: 1.2,
        tier: "tier3",
        iconClass: "fa-solid fa-children",
        gridPos: 21
    },
    {
        id: "radio-tv",
        name: "Radio and TV Stations",
        multiplier: 1.2,
        tier: "tier3",
        iconClass: "fa-solid fa-tower-broadcast",
        gridPos: 22
    },
    {
        id: "baby-accessories",
        name: "Baby Clothes, Accessories & Toys",
        multiplier: 1.1,
        tier: "tier3",
        iconClass: "fa-solid fa-baby",
        gridPos: 23
    },
    {
        id: "accounting-tax",
        name: "Accounting & Tax Services",
        multiplier: 1.1,
        tier: "tier3",
        iconClass: "fa-solid fa-chart-pie",
        gridPos: 24
    },
    {
        id: "expansion-slot",
        name: "Expansion Slot",
        multiplier: 1.0,
        tier: "tier3",
        iconClass: "fa-solid fa-plus",
        gridPos: 25,
        isPlaceholder: true
    }
];

export const PLATFORM2_TIERS = {
    tier1: {
        name: "High Value",
        multiplier: "2.0x - 2.5x",
        color: "from-amber-400 to-orange-500",
        shadow: "shadow-orange-500/20"
    },
    tier2: {
        name: "Medium Value",
        multiplier: "1.4x - 1.8x",
        color: "from-blue-400 to-indigo-500",
        shadow: "shadow-indigo-500/20"
    },
    tier3: {
        name: "Entry Level",
        multiplier: "1.0x - 1.3x",
        color: "from-emerald-400 to-teal-500",
        shadow: "shadow-teal-500/20"
    }
};

File: backend/app/platform2_categories.py
"""
platform2_categories.py — Second Advertiser Platform Category Configuration.

This module defines the category structure for the broader B2C/B2B advertising platform.
It operates independently from the existing Automotive B2B categories in industry_config.py.

Architecture:
    - 24 categories organised into 3 tiers (High Value, Medium Value, Entry Level)
    - 1 reserved slot for easy future addition (total = 25 for 5×5 grid)
    - Each category has a multiplier used in the pricing formula:
        final_price = base_price × category_multiplier × ad_format_multiplier × coverage_multiplier

Adding a new category:
    1. Add it to the appropriate TIER dict below
    2. The frontend grid in platform2Categories.js must also be updated
    3. The system supports up to 25 categories for the 5×5 grid layout
"""

from typing import Dict, Optional, List

# ─── Base price for all Platform 2 categories ───
PLATFORM2_BASE_PRICE: float = 100.0


# ─── TIER 1 — High Value ───
TIER1_HIGH_VALUE: Dict[str, float] = {
    "Real Estate & Property Agents": 2.5,
    "Legal Services, Lawyers & Mediation": 2.5,
    "Financial and Insurance Services": 2.3,
    "Health, Wellness & Medical": 2.3,
    "Automotive Services": 2.2,
    "IT & Tech Support Services": 2.0,
    "Professional Training & Certification": 2.0,
    "Department Stores and Electronics": 2.0,
    "Mobile Phone and Internet Services": 2.0,
}

# ─── TIER 2 — Medium Value ───
TIER2_MEDIUM_VALUE: Dict[str, float] = {
    "Education & Tutoring": 1.8,
    "Event & Wedding Services": 1.8,
    "Beauty and Cosmetic Surgery": 1.7,
    "Fitness & Personal Training": 1.6,
    "Home & Garden": 1.6,
    "Lifestyle, Boutique, Apparel & Accessories": 1.5,
    "Travel & Tourism": 1.5,
    "Storage, Logistics and Removalists": 1.5,
}

# ─── TIER 3 — Entry Level ───
TIER3_ENTRY_LEVEL: Dict[str, float] = {
    "Restaurants, Food & Beverage": 1.3,
    "Trades & Home Services": 1.3,
    "Pets & Animals": 1.2,
    "Childcare & Aged Care Providers": 1.2,
    "Radio and TV Stations": 1.2,
    "Baby Clothes, Accessories & Toys": 1.1,
    "Accounting & Tax Services": 1.1,
}

# ─── Combined category multiplier registry ───
PLATFORM2_CATEGORY_MULTIPLIERS: Dict[str, float] = {
    **TIER1_HIGH_VALUE,
    **TIER2_MEDIUM_VALUE,
    **TIER3_ENTRY_LEVEL,
}

# ─── Tier metadata (for frontend display) ───
PLATFORM2_TIERS = {
    "tier1": {
        "name": "Tier 1 — High Value",
        "description": "Premium categories with highest advertiser lifetime value",
        "categories": list(TIER1_HIGH_VALUE.keys()),
    },
    "tier2": {
        "name": "Tier 2 — Medium Value",
        "description": "Mid-range categories with strong demand",
        "categories": list(TIER2_MEDIUM_VALUE.keys()),
    },
    "tier3": {
        "name": "Tier 3 — Entry Level",
        "description": "Accessible categories with competitive pricing",
        "categories": list(TIER3_ENTRY_LEVEL.keys()),
    },
}

# ─── Grid layout: ordered list for 5×5 display ───
# This defines the exact order categories appear in the frontend grid.
# Position 25 is reserved for future expansion.
PLATFORM2_GRID_ORDER: List[str] = [
    # Row 1
    "Real Estate & Property Agents",
    "Legal Services, Lawyers & Mediation",
    "Financial and Insurance Services",
    "Health, Wellness & Medical",
    "Automotive Services",
    # Row 2
    "IT & Tech Support Services",
    "Professional Training & Certification",
    "Department Stores and Electronics",
    "Mobile Phone and Internet Services",
    "Education & Tutoring",
    # Row 3
    "Event & Wedding Services",
    "Beauty and Cosmetic Surgery",
    "Fitness & Personal Training",
    "Home & Garden",
    "Lifestyle, Boutique, Apparel & Accessories",
    # Row 4
    "Travel & Tourism",
    "Storage, Logistics and Removalists",
    "Restaurants, Food & Beverage",
    "Trades & Home Services",
    "Pets & Animals",
    # Row 5
    "Childcare & Aged Care Providers",
    "Radio and TV Stations",
    "Baby Clothes, Accessories & Toys",
    "Accounting & Tax Services",
    # Slot 25: reserved — add your new category here
]


def get_platform2_category_multiplier(category_name: str, default: float = 1.0) -> float:
    """
    Return the multiplier for a Platform 2 category.
    Falls back to `default` if the category is not found.
    """
    return PLATFORM2_CATEGORY_MULTIPLIERS.get(category_name, default)


def get_platform2_category_price(category_name: str) -> float:
    """
    Return the category-level price: base_price × category_multiplier.
    """
    multiplier = get_platform2_category_multiplier(category_name)
    return PLATFORM2_BASE_PRICE * multiplier


def get_platform2_all_categories() -> List[dict]:
    """
    Return all categories with their tier and multiplier, ordered for the 5×5 grid.
    """
    tier_lookup = {}
    for tier_key, tier_data in PLATFORM2_TIERS.items():
        for cat in tier_data["categories"]:
            tier_lookup[cat] = {
                "tier_key": tier_key,
                "tier_name": tier_data["name"],
            }

    result = []
    for idx, cat_name in enumerate(PLATFORM2_GRID_ORDER):
        tier_info = tier_lookup.get(cat_name, {"tier_key": "tier3", "tier_name": "Tier 3 — Entry Level"})
        result.append({
            "name": cat_name,
            "multiplier": PLATFORM2_CATEGORY_MULTIPLIERS.get(cat_name, 1.0),
            "tier": tier_info["tier_key"],
            "tier_name": tier_info["tier_name"],
            "grid_position": idx + 1,
            "base_price": PLATFORM2_BASE_PRICE,
            "category_price": PLATFORM2_BASE_PRICE * PLATFORM2_CATEGORY_MULTIPLIERS.get(cat_name, 1.0),
        })

    return result
